quantum/keyboard.c

```diff
@@ -252,6 +252,14 @@ __attribute__((weak)) void keyboard_post_init_kb(void) {
     keyboard_post_init_user();
 }
 
+/** \brief is_matrix_locked
+ *
+ * FIXME: needs doc
+ */
+__attribute__((weak)) bool is_matrix_locked(void) {
+    return false;
+}
+
```

Comment:
Maybe this should be a proper part of the matrix API?
- rename to `matrix_is_locked()`
- add some actual description instead of FIXME
- add a prototype to `quantum/matrix.h`
- the implementation might need to stay here, because even `quantum/matrix_common.c` is skipped when using a “full” custom matrix

Also I'm not sure about the “is_locked” name — the usage here is more like `matrix_available()` (with the logic reversed from the current one), similar to `uart_available()` for the nonblocking receive API.